src/ram-lib-mapping.ts

```diff
@@ -0,0 +1,532 @@
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  // update num tokens in auction detail in project minter configuration extraMinterDetails json field
+  let auctionConfigUpdateDetails: TypedMap<
+    string,
+    JSONValue
+  > = createTypedMapFromEntries([
+    {
+      key: "auctionNumTokensInAuction",
+      value: toJSONValue(event.params.numTokensInAuction.toString())
+    }
+  ]);
+
+  mergeProjectMinterConfigExtraMinterDetails(
+    projectMinterConfig,
+    auctionConfigUpdateDetails
+  );
+
+  projectMinterConfig.save();
+
+  // induce sync if the project minter configuration is the active one
+  updateProjectIfMinterConfigIsActive(
+    minterProjectAndConfig.project,
+    projectMinterConfig,
+    event.block.timestamp
+  );
+}
+
+export function handleBidCreated(event: BidCreated): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  // Get the auction base price
+  const currentProjectMinterConfigDetails = getProjectMinterConfigExtraMinterDetailsTypedMap(
+    projectMinterConfig
+  );
+  const auctionBasePrice = currentProjectMinterConfigDetails.get(
+    "auctionBasePrice"
+  );
+
+  if (!auctionBasePrice) {
+    log.warning(
+      "Auction base price for minter {} not found for core contract {}",
+      [event.address.toString(), event.params.coreContract.toHexString()]
+    );
+    return;
+  }
+  const bidValue = slotIndexToBidValue(
+    auctionBasePrice.toString(),
+    event.params.slotIndex
+  );
+
+  const fullProjectId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.projectId
+  );
+
+  const ramBidId = generateRAMMinterBidId(
+    event.address.toHexString(),
+    fullProjectId,
+    event.params.bidId.toString()
+  );
+  // @dev: a bid with this ID should not already exist for the RAM Minter
+  const bid = new Bid(ramBidId);
+  if (!bid) {
+    log.warning("Bid ID {} not found for minter {}", [
+      ramBidId,
+      event.address.toHexString()
+    ]);
+    return;
+  }
+  // Create new account entity if one for the bidder doesn't exist
+  const bidderAccount = new Account(event.params.bidder.toHexString());
+  bidderAccount.save();
+
+  bid.project = minterProjectAndConfig.project.id;
+  bid.minter = event.address.toHexString();
+  bid.value = bidValue;
+  bid.bidder = bidderAccount.id;
+  bid.settled = false;
+  bid.slotIndex = event.params.slotIndex;
+  bid.winningBid = true;
```

Comment:
Do we want winning bid to default to true? I feel like it should either align with the `settled` field (maybe a bit redundant) or made optional. 

Is there any case where winningBid and settled should have different values? I sort of imagine settled to be the equivalent of SEA winning bid but could be misunderstanding. WDYT?

Reply:
Sweet sounds good to keep! Thinking a bit more, it might make sense to have it be not optional and default to false instead of null to make it marginally easier to work with (no need to cast to a boolean if a function explicitly requires one) but ultimately ver minor difference so don't feel too strong.

Reply:
Awesome, that makes sense to me! Let's do that - updated the logic + tests 🙏